src/components/Search.tsx

```diff
@@ -0,0 +1,102 @@
+'use client';
+
+import { useState } from 'react';
+import useDebouncedEffect from '../lib/useDebouncedEffect';
+import { API_DOMAIN } from '../lib/constants';
+import { SearchResuts } from '../types';
+import Column from './Column';
+import Row from './Row';
+
+async function fetchResults(query) {
+  const data: SearchResuts = await fetch(`${API_DOMAIN}/api/v2/search?q=${query}`, {
+    cache: 'no-cache', // seconds
+  }).then((res) => res.json());
+
+  return data;
+}
+
+const Search = () => {
+  const [data, setData] = useState<SearchResuts | null>(null);
+  const [activeFilter, setActiveFilter] = useState<string>('artists');
+  const [searchValue, setSearchValue] = useState<string>('');
+
+  useDebouncedEffect(
+    () => {
+      // TODO: sanitize, trim ...
+      if (searchValue) {
+        fetchResults(searchValue).then(setData);
+      } else {
+        setData(null);
```

Comment:
Currently it shows loading before user has typed anything, which may not be ideal